packages/ui-library/src/theme/variants/toggle-button-group-variants.ts

```diff
@@ -14,19 +14,20 @@ export const buildToggleButtonGroupVariant = (mode: 'light' | 'dark'): Component
   MuiToggleButton: {
     styleOverrides: {
       root: {
-        minWidth: SPACING(24),
         padding: `${SPACING(2.5)} ${SPACING(3)}`,
         textTransform: 'none',
-        borderRadius: `${SPACING(2)} !important`,
-        border: `1px solid ${colors[mode].border.border1} !important`,
         transition: 'background-color 300ms, border-color 300ms',
         background: colors[mode].background.secondary,
-        ':hover': {
-          background: colors[mode].background.tertiary,
+        '&&&.MuiToggleButton-root': {
+          border: `1px solid ${colors[mode].border.border1}`,
+          borderRadius: `${SPACING(2)}`,
+          ':hover': {
+            background: colors[mode].background.tertiary,
+          },
         },
-        '&.Mui-selected': {
+        '&&&.Mui-selected': {
```

Comment:
those are a lot of &, why do you need those? should only be one

Reply:
There's probably a selected style override you can set this there

Reply:
About the specificity, the selector used by MUI is something like `.ktEYlR .MuiToggleButtonGroup-grouped:not(:last-of-type) {`. Having one & isn't being enought. I have downgrade this to && though 😄,

The `selected` is not working, in fact throws an error in the console when using it

Reply:
In order to override border+borderRadius properties, I needed a more specific selector. The one used by MUI is something like `.ktEYlR .MuiToggleButtonGroup-grouped:not(:last-of-type) {`. Having one `&` isn't being enought, so to override this I needed to use `&&&`.
Then, this implied that for `.Mui-selected` case I needed to use a selector with same specificity.

The selected rule is not working, in fact throws an error in the console when using it, suggesting to use .Mui-selected

Reply:
Right, take a look into the Css classes part of https://mui.com/material-ui/api/button-group/

Just like we target `root`, you can target `contained` and etc

Reply:
I couldn't find a working prop for this case, but I was able to improve this specificity issue